insights/parsers/pmrep.py

```diff
@@ -0,0 +1,80 @@
+"""
+Pmrep - command ``pmrep -t 1s -T 1s network.interface.out.packets network.interface.collisions swap.pagesout``
+==============================================================================================================
+
+Parse the content of the ``pmrep -t 1s -T 1s network.interface.out.packets network.interface.collisions swap.pagesout`` command.
+
+Sample ``pmrep -t 1s -T 1s network.interface.out.packets network.interface.collisions swap.pagesout`` command output::
+
+    n.i.o.packets  n.i.o.packets  n.i.collisions  n.i.collisions  s.pagesout
+             eth0             lo            eth0              lo
+          count/s        count/s         count/s         count/s     count/s
+              N/A            N/A             N/A             N/A         N/A
+            1.997          0.000           0.000           0.000       0.000
+
+Examples:
+    >>> type(pmrep_doc_obj)
+    <class 'insights.parsers.pmrep.PMREPMetrics'>
+    >>> pmrep_doc_obj.data.get('n.i.o.packets', None)
+    [{'eth0': '1.000'}, {'lo': '2.000'}]
+    >>> pmrep_doc_obj.data.get('n.i.collisions', None)
+    [{'eth0': '3.000'}, {'lo': '4.000'}]
+    >>> pmrep_doc_obj.data.get('s.pagesout', None)
+    ['5.000']
+"""
+
+from insights import Parser, parser
+from insights.specs import Specs
+from insights.parsers import SkipException
+
+
+@parser(Specs.pmrep_metrics)
+class PMREPMetrics(Parser):
+    """Parses output of ``pmrep -t 1s -T 1s network.interface.out.packets network.interface.collisions swap.pagesout`` command."""
+    def parse_content(self, content):
+        if not content:
+            raise SkipException("Empty content.")
+
+        heading_keys = ["n.i.o.packets", "n.i.collisions", "s.pagesout"]
```

Comment:
The [parse_fixed_table](https://github.com/RedHatInsights/insights-core/blob/master/insights/parsers/__init__.py#L282) would be helpful to simplify the logic of this parse_content.

Reply:
@xiangce - I think using the "parse_fixed_table" method will not be helpful in this case because this method overrides the column name when we have multiple columns with the same name. Below is the example of it:

```
  n.i.o.packets  n.i.o.packets  n.i.collisions  n.i.collisions  s.pagesout
           eth0             lo            eth0              lo
        count/s        count/s         count/s         count/s     count/s
            N/A            N/A             N/A             N/A         N/A
          1.000          2.000           3.000           4.000       5.000
```

- Using the "parse_fixed_table" method, as we have "n.i.o.packets" and "n.i.o.packets" column name twice, so it overides for the "eth0" row and only uses for "lo".
```
[{'n.i.o.packets': 'lo', 'n.i.collisions': 'lo', 's.pagesout': ''}, {'n.i.o.packets': 'count/s', 'n.i.collisions': 'count/s', 's.pagesout': 'count/s'}, {'n.i.o.packets': 'N/A', 'n.i.collisions': 'N/A', 's.pagesout': 'N/A'}, {'n.i.o.packets': '2.000', 'n.i.collisions': '4.000', 's.pagesout': '5.000'}]
```